pkgs/development/tools/build-managers/buck/default.nix

```diff
@@ -0,0 +1,46 @@
+  { stdenv, pkgs, fetchFromGitHub, fetchgit, jdk, ant, python2, watchman, unzip, bash }:
+
+stdenv.mkDerivation rec {
+  name = "buck-${version}";
+  version = "v2017.05.31.01";
+
+  src = fetchFromGitHub {
+    owner = "facebook";
+    repo = "buck";
+    rev = "0b8b3828a11afa79dc128832cb55b106f07e48aa";
+    sha256 = "1g3yg8qq91cdhsq7zmir7wxw3767l120f5zhq969gppdw9apqy0s";
+  };
+
+  patches = [ ./pex-mtime.patch ];
+
+  postPatch = ''
+    for f in $(grep -l -r '/bin/bash'); do
+      substituteInPlace "$f" --replace '/bin/bash' '${bash}/bin/bash'
```

Comment:
Do you only replace shebangs here? If so, add bash as input and use `patchShebangs .`.